core/src/test/java/org/mobilitydata/gtfsvalidator/table/AnyTableLoaderTest.java

```diff
@@ -0,0 +1,216 @@
+import org.mobilitydata.gtfsvalidator.validator.ValidatorLoader;
+import org.mobilitydata.gtfsvalidator.validator.ValidatorProvider;
+
+public class AnyTableLoaderTest {
+
+  ValidationContext validationContext;
+
+  @Before
+  public void setup() {
+    validationContext =
+        ValidationContext.builder()
+            .setCountryCode(CountryCode.forStringOrUnknown("CA"))
+            .setCurrentDateTime(new CurrentDateTime(ZonedDateTime.now(ZoneId.systemDefault())))
+            .build();
+  }
+
+  @Test
+  public void invalidInputStream() {
+    var testTableDescriptor = mock(GtfsTableDescriptor.class);
+    when(testTableDescriptor.gtfsFilename()).thenReturn("_not_a_valid_file_");
+    NoticeContainer loaderNotices = new NoticeContainer();
+
+    AnyTableLoader.load(
+        testTableDescriptor,
+        mock(ValidatorProvider.class),
+        retrieveInputStream("_not_a_valid_file_"),
+        loaderNotices);
+
+    assertTrue(loaderNotices.hasValidationErrors());
+
+    assertValidationNotice(loaderNotices, CsvParsingFailedNotice.class);
+    assertEquals(1, loaderNotices.getValidationNotices().size());
+    verify(testTableDescriptor, times(0)).createContainerForHeaderAndEntities(any(), any(), any());
+  }
+
+  @Test
+  public void emptyInputStream() {
+    var testTableDescriptor = mock(GtfsTableDescriptor.class);
+    when(testTableDescriptor.gtfsFilename()).thenReturn("filename");
+    NoticeContainer loaderNotices = new NoticeContainer();
+    InputStream csvInputStream = new ByteArrayInputStream("".getBytes(StandardCharsets.UTF_8));
+    ValidatorProvider validatorProvider = mock(ValidatorProvider.class);
+
+    AnyTableLoader.load(testTableDescriptor, validatorProvider, csvInputStream, loaderNotices);
+
+    assertTrue(loaderNotices.hasValidationErrors());
+    assertEquals(1, loaderNotices.getValidationNotices().size());
+    assertThat(loaderNotices.getValidationNotices())
+        .containsExactly(new EmptyFileNotice("filename"));
+    verify(testTableDescriptor, times(0)).createContainerForHeaderAndEntities(any(), any(), any());
+    verify(validatorProvider, times(0)).createSingleFileValidators(any());
+  }
+
+  @Test
+  public void invalidHeaders() {
+    var testTableDescriptor = mock(GtfsTableDescriptor.class);
+    when(testTableDescriptor.gtfsFilename()).thenReturn("filename");
+    GtfsColumnDescriptor[] columns =
+        new GtfsColumnDescriptor[] {
+          AutoValue_GtfsColumnDescriptor.builder()
+              .setColumnName("route_id")
+              .setHeaderRequired(true)
+              .setFieldLevel(FieldLevelEnum.REQUIRED)
+              .setIsMixedCase(false)
+              .setIsCached(false)
+              .build()
+        };
+    when(testTableDescriptor.getColumns()).thenReturn(ImmutableList.copyOf(columns));
+    ValidatorProvider validatorProvider =
+        spy(
+            new DefaultValidatorProvider(
+                mock(ValidationContext.class), mock(ValidatorLoader.class)));
```

Comment:
Why do you need to spy on this?

Reply:
So no longer necessary to mock it?

Also, you seem to use different combinations of mocked ValidatorProvider vs DefaultValidatorProvider here and in other tests.  Possible just to use DefaultValidatorProvider, possibly created in @Before section along with context?